app/models/school.rb

```diff
@@ -67,4 +67,12 @@ def can_order_devices?(device_type = 'std_device')
     allocation = device_allocations.by_device_type(device_type).first
     allocation&.cap.to_i > allocation&.devices_ordered.to_i
   end
+
+  def invite_school_contact_if_possible!
```

Comment:
just one minor nit-pick about this name - having `(...)_if_possible` combined with the bang feels inconsistent, like the bang implies it will raise an exception if it's not possible rather than return false (in line with the usual Rails `update` vs `update!` convention). I know there's the _other_ convention at play, that the bang means it's modifying internal state (which it is), so maybe it's just me :/

Reply:
I'll drop the `if_possible!` from the method name because I agree it doesn't add much and looks strange.

What would you suggest doing about the bang? I could remove it since the method doesn't throw exceptions, and the name should be clear enough that it's state-modifying...

Reply:
Yeah, TBH I think just `invite_school_contact` does the trick

Reply:
OK, as discussed on Slack, I've renamed the method to `invite_school_contact` in 7aa1c3a